Sources/Classes/RSDeviceInfo.m

```diff
@@ -9,23 +9,46 @@
 #import "RSDeviceInfo.h"
 #import "RSContext.h"
 #import <UIKit/UIKit.h>
+#if TARGET_OS_WATCH
+#import <WatchKit/WKInterfaceDevice.h>
+#endif
+
 
 @implementation RSDeviceInfo
 
 - (instancetype)init
 {
     self = [super init];
+
     if (self) {
+#if !TARGET_OS_WATCH
         _identifier = [[[[UIDevice currentDevice] identifierForVendor] UUIDString]lowercaseString];
-        _manufacturer = @"Apple";
         _model = [[UIDevice currentDevice] model];
         _name = [[UIDevice currentDevice] name];
-        _type = @"iOS";
+#else
+        _identifier = [[[[WKInterfaceDevice currentDevice] identifierForVendor]UUIDString] lowercaseString];
+        _model = [[WKInterfaceDevice currentDevice]model];
+        _name = [[WKInterfaceDevice currentDevice]name];
+#endif
+        _manufacturer = @"Apple";
         _attTrackingStatus = RSATTNotDetermined;
+        _type = [self getDeviceType];
     }
+
     return self;
 }
 
+- (NSString *) getDeviceType {
+#if TARGET_OS_WATCH
```

Comment:
Where is iPadOS?

Reply:
Actually, we cannot check if the app is running on iPadOS at compile time. We don't have something like `TARGET_OS_IPAD`. 

So setting `type` the same way we are setting `name` in RSDeviceInfo.m